processor.gradle

```diff
@@ -5,15 +5,20 @@ version = projectVersion
 
 apply from: 'https://raw.githubusercontent.com/carsnip/common-build/master/common.gradle'
 
-task buildDocker(type: Exec) {
-  workingDir projectDir
-  commandLine 'sh', '-c', "docker/build.sh $projectName $projectVersion $projectMainClass"
-}
+String jolokiaPort = "7777"
+String dockerRepo = "963107833485.dkr.ecr.eu-west-1.amazonaws.com"
+String dockerTag = project.hasProperty('branchName') && project.hasProperty('branchVersion') ? "${project.property('branchName')}-${project.property('branchVersion')}" : "test"
 
-buildDocker.dependsOn(distZip)
-
-jar {
-  manifest {
-    attributes 'Main-Class': mainClassName
+jib {
+  from {
+    image = "$dockerRepo/docker-java-11.0.5-jre:master-11".toString
+  }
+  to {
+    image = "$dockerRepo/$projectName:$dockerTag".toString()
+  }
+  container {
+    jvmFlags = ['-Xmx1024m', '-XX:+ExitOnOutOfMemoryError', "-javaagent:/opt/jolokia.jar=port=${jolokiaPort},host=*".toString()]
```

Comment:
Can this jolokiaPort to set as a ENV in the Docker image, defaulting to 7777? It can then be overridden at run time and the default can always set to 7777.